R/cor_df.R

```diff
@@ -192,8 +192,16 @@ network_plot.cor_df <- function(rdf,
   rdf <-  as_matrix(rdf, diagonal = 1)
   distance <- 1 - abs(rdf)
 
-  # Use multidimensional Scaling to obtain x and y coordinates for points.
-  points <- suppressWarnings(stats::cmdscale(distance))
+  points <- if (ncol(rdf) == 1) {
+    # 1 var: a single central point
+    matrix(c(0, 0), ncol = 2, dimnames = list(colnames(rdf)))
+  } else if (ncol(rdf) == 2) {
+    # 2 vars: 2 opposing points
+    matrix(c(0, -0.1, 0, 0.1), ncol = 2, dimnames = list(colnames(rdf)))
+  } else {
+    # More than 2 vars: multidimensional scaling to obtain x and y coordinates for points.
+    suppressWarnings(stats::cmdscale(distance, k = 2))
+  }
```

Comment:
Instead of assigning to an `if()` statement, can you use `switch()` here?
https://adv-r.hadley.nz/control-flow.html#switch

Reply:
That's a good idea but not feasible here as there is no way to set default value when switching on a numeric variable.

For example, the following does not work as it will set points to NULL when `ncol(rdf) > 3`:

```r
points <- switch(
    ncol(rdf),
    # 1 var: a single central point
    matrix(c(0, 0), ncol = 2, dimnames = list(colnames(rdf))),
    # 2 vars: 2 opposing points
    matrix(c(0, -0.1, 0, 0.1), ncol = 2, dimnames = list(colnames(rdf))),
    # More than 2 vars: multidimensional scaling to obtain x and y coordinates for points.
    suppressWarnings(stats::cmdscale(distance, k = 2))
  )
```

You can set a default when switching on a character variable but I don't think that's what you meant as it is pretty ugly:

```r
points <- switch(
    as.character(ncol(rdf)),
    # 1 var: a single central point
    "1" = matrix(c(0, 0), ncol = 2, dimnames = list(colnames(rdf))),
    # 2 vars: 2 opposing points
    "2" = matrix(c(0, -0.1, 0, 0.1), ncol = 2, dimnames = list(colnames(rdf))),
    # More than 2 vars: multidimensional scaling to obtain x and y coordinates for points.
    suppressWarnings(stats::cmdscale(distance, k = 2))
  )
```

Reply:
Ah right, I forgot that there is no default value allowed for numeric expressions. 👍